docs/configuration.asciidoc

```diff
@@ -692,6 +692,18 @@ The timeout is applied once the agent has sent the entire request body to the AP
 If the response from the server takes longer than allowed by this timeout,
 the HTTP request is terminated and the TCP socket closed.
 
+[[sanitize-field-names]]
+==== `sanitizeFieldNames`
+* *Type:* Array
+* *Default:* ['password', 'passwd', 'pwd', 'secret', '\*key', '*token*', '\*session*', '\*credit*', '\*card*', 'authorization', 'set-cookie']
```

Comment:
@bmorelli25 do you know how asciidoc `*` escaping works? I tried escaping each individual item but that ended up with a mismash of bolding.  The above seems to work, but seems odd that `token` doesn't need escaping.

Reply:
Youch, that's unfortunate.  Frankly that feels like a bug in asciidoc design (or at least a wart), but I don't know it well enough to say.

How about this:

```
* *Default:* +['password', 'passwd', 'pwd', 'secret', '*key', '*token*', '*session*', '*credit*', '*card*', 'authorization', 'set-cookie']+
```

https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#escaping-text

Reply:
If you turn the entire array into a code snippet, there's no escaping required:

```suggestion
* *Default:* `['password', 'passwd', 'pwd', 'secret', '*key', '*token*', '*session*', '*credit*', '*card*', 'authorization', 'set-cookie']`
```

<img width="643" alt="Screen Shot 2020-12-07 at 11 20 56 AM" src="https://user-images.githubusercontent.com/5618806/101395086-4cc5d380-387e-11eb-9a5a-fe81851f9766.png">

Reply:
Brandon, are you sure that works with asciidoc in general? Here is my attempt using GH's rendering of asciidoc showing all the forms we've tried here:
https://gist.github.com/trentm/3bf4bd9473c40dfae7d2221058a26bf1

Note how the "Default (with backticks per Brandon)" case seems to "use" the asterisk on "key" to start bolding -- at least with GH's rendering.

Reply:
> Brandon, are you sure that works with asciidoc in general?

Nope, but I'm not surprised that it doesn't. I only tested in our doc build; our implementation of asciidoc doesn't always behave. . . as expected?

Here's what I tried and how it renders in our system:

```
* *Default:* `['password', 'passwd', 'pwd', 'secret', '*key', '*token*', '*session*', '*credit*', '*card*', 'authorization', 'set-cookie']`

* *Default:* ['password', 'passwd', 'pwd', 'secret', '\*key', '*token*', '\*session*', '\*credit*', '\*card*', 'authorization', 'set-cookie']

* *Default:* +['password', 'passwd', 'pwd', 'secret', '*key', '*token*', '*session*', '*credit*', '*card*', 'authorization', 'set-cookie']+

* *Default:* `[+password+, +passwd+, +pwd+, +secret+, +\*key+, +\*token*+, +\*session*+, +\*credit*+, +\*card*+, +authorization+, +set-cookie+]`
```

<img width="676" alt="Screen Shot 2020-12-07 at 2 28 09 PM" src="https://user-images.githubusercontent.com/5618806/101413141-9de2c100-3898-11eb-842d-39ae1f065ef7.png">

Reply:
Oh! Also, our build system uses Asciidoctor, not Asciidoc. The syntax is mostly the same, but there are quite a few [differences](https://asciidoctor.org/docs/asciidoc-asciidoctor-diffs/). That's probably the main reason why these are rendering so different.

Reply:
Whoa. TIL those are two different things.

Reply:
Yeah, that's likely my fault, sorry! I tend to use the two terms interchangeably, which isn't correct.

Reply:
Those diffs don't mention the difference in behaviour of `+foo+` for "passthrough", https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#escaping-text

And the https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#formatted-text section seems to indicate that backticks shouldn't be escaping, only styling as monospace. 

https://asciidoctor.org/docs/migration/#new-syntax refers to changes in backtick and plus handling in asciidoctor v1.5.0. I'm a little confused, but likely not a big deal. :)

Reply:
So, yeah, good point. While asciidoctor and asciidoc are different, that doesn't appear to be the reasoning behind these inconsistencies. Our implementation of asciidoctor is highly customized and while some things work the same way, some don't. Consider the https://asciidoctor.org/ reference a great starting point, but not a source of truth. Our source of truth is the https://github.com/elastic/docs readme file. The only downside is it's not as exhaustive as the asciidoctor reference.

Reply:
Thanks for the link. That helps.

Reply:
@bmorelli25 The "treat this as code" meta-charaters worked great _and_ match our handling of the other configuration values.  👍